File: src/nfan/fan/sys/Param.cs
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   23 Jul 07  Andy Frank  Creation
//

namespace Fan.Sys
{
  /// <summary>
  /// Param represents one parameter definition of a Func (or Method).
  /// </summary>
  public class Param : FanObj
  {

  //////////////////////////////////////////////////////////////////////////
  // Constructor
  //////////////////////////////////////////////////////////////////////////

    public Param(Str name, Type of, int mask)
    {
      this.m_name = name;
      this.m_of   = of;
      this.m_mask = mask;
    }

  //////////////////////////////////////////////////////////////////////////
  // Methods
  //////////////////////////////////////////////////////////////////////////

    public override Type type() { return Sys.ParamType; }

    public Str name()  { return m_name; }
    public Type of()   { return m_of; }
    public Bool hasDefault() { return Bool.make(m_mask & HAS_DEFAULT); }

    public override Str toStr() { return Str.make(m_of + " " + m_name); }

  //////////////////////////////////////////////////////////////////////////
  // Fields
  //////////////////////////////////////////////////////////////////////////

    public static readonly int HAS_DEFAULT = 0x01;  // is a default value provided

    internal readonly Str m_name;
    internal readonly Type m_of;
    internal readonly int m_mask;
  }
}

File: src/jfan/fan/sys/Param.java
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   4 Jan 06  Brian Frank  Creation
//
package fan.sys;

/**
 * Param represents one parameter definition of a Func (or Method).
 */
public class Param
  extends FanObj
{

//////////////////////////////////////////////////////////////////////////
// Constructor
//////////////////////////////////////////////////////////////////////////

  public Param(Str name, Type of, int mask)
  {
    this.name = name;
    this.of   = of;
    this.mask = mask;
  }

//////////////////////////////////////////////////////////////////////////
// Methods
//////////////////////////////////////////////////////////////////////////

  public Type type() { return Sys.ParamType; }

  public Str name()  { return name; }
  public Type of()   { return of; }
  public Boolean hasDefault() { return (mask & HAS_DEFAULT) != 0; }

  public Str toStr() { return Str.make(of + " " + name); }

//////////////////////////////////////////////////////////////////////////
// Fields
//////////////////////////////////////////////////////////////////////////

  public static final int HAS_DEFAULT   = 0x01;  // is a default value provided

  final Str name;
  final Type of;
  final int mask;

}